### PR TITLE
First draft of Quickstart guide

### PR DESCRIPTION
**Pull request by melissawm**
This also includes a basic outline of the documentation (yet to be written).

### CODE REVIEW

File: docs/quickstart.rst [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  >>> with versioned_file.stage_version('version2') as group:
+  ...     group['mydataset'] = np.ones(10000)
+
+The context manager returns a h5py *group* object, which should be modified in-place to build the new version. When the context manager exits, the version will be written to the file. This has two effects. First, the h5py file object ``fileobject`` now has metadata associated with versions::
+
+  >>> fileobject.keys()
+  <KeysViewHDF5 ['_version_data']>
+
+You can check that ``fileobject['_version_data']`` is a HDF5 group. But most importantly, ``versioned_file`` can now be used to expose versioned data directly, by version name::
```
Comment (asmeurer) [inline docs/quickstart.rst:33]: I'm not sure if we should talk about the `_version_data` internals too much here. I guess we can mention that all the data from the versioned hdf5 is stored in the `_version_data` group on the file, and that any interaction with the versioning should happen through the versioned_hdf5 api.

File: docs/quickstart.rst [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  >>> dataset = v2['mydataset']
+  >>> dataset[()]
+  array([1., 1., 1., ..., 1., 1., 1.])
+
+.. note::
+
+   Versioned HDF5 files have a special structure and should not be modified directly. Also note that once a version is created in the file, it should be treated as read-only. Some protections are in place to prevent accidental modification, but it is not possible in the HDF5 layer to make a dataset or group read-only, so modifications made outside of this library could result in breaking things.
+
+When you are done manipulating data, use ``fileobject.close()`` to make sure the HDF5 file is written properly to disk. Note that the ``VersionedHDF5File`` object does not need to be closed.
```
Comment (asmeurer) [inline docs/quickstart.rst:51]: I wonder if it would make sense to add a `close()` method to `VersionedHDF5File` that closes the file.

**Comment by asmeurer** [comment]
This looks good to me. You can merge this right away if you want or if you want to wait until you have written more that is fine too. Once you merge it it will automatically be pushed to https://quansight.github.io/versioned-hdf5/

File: docs/quickstart.rst [hunk 3]
```diff
@@ -0,0 +1,51 @@
+Quickstart Guide
+================
+
+Let's say you have an HDF5 file with contents that might change over time. You may add or remove datasets, change the contents of the data or the metadata, and would like to keep a record of which changes occurred when, and a way to recover previous versions of this file. Versioned HDF5 allows you to do that by building a versioning API on top of h5py.
```
Comment (scopatz) [inline docs/quickstart.rst:4]: Please limit lines to ~80 chars

File: docs/quickstart.rst (hunk 2 above)
Comment (scopatz) [reply docs/quickstart.rst:51]: It probably does

**Comment by scopatz** [comment]
Thanks for putting this in @melissawm!

**Pull request merged**